### ISSUE TITLE
[Question] Download an individual file

### ISSUE THREAD

**Issue by ajthemacboy** [jpillora/cloud-torrent#39]
Is it possible to download a single, individual file? I want to download a sample file without downloading the full torrent, and the sample file is only one of several files in the torrent.

Thanks.

**Comment by jpillora** [comment]
This feature is coming :)

**Comment by ajthemacboy** [comment]
@jpillora Awesome, I'll be waiting :)

**Comment by sponde25** [comment]
Hi, any update on this feature? This would make life really simple. 

BTW thanks for sharing this amazing work with us.

**Comment by jpillora** [comment]
Can't afford to put in the time just yet, so can't give an ETA yet. Since many new features all interrelate, I plan on doing them all at once in https://github.com/jpillora/cloud-torrent/milestones/0.9. There's quite large changes to the way files are stored and written which then leads to more big changes so it's quite a task.

**Comment by atahanguner** [comment]
I want a download all files for zip
How can i do this?

**Comment by squarefractal** [comment]
I know that this is scheduled for 0.9, however, can you please give this more priority over the other planned features?

I think there are many people who would want to select specific files to download instead of the entire torrent, which can be quite large at times.

**Comment by jpillora** [comment]
Agreed, it would be nice. The [torrent library](https://github.com/anacrolix/torrent) has been improved and it would be now easier to implement. However, my day job is currently taking all my time at the moment though so I've had to put my open-source projects on pause for a while.

**Comment by shogunreaper** [comment]
any update on this?

**Comment by jpillora** [comment]
Doesn't resolve this issue, though for others looking downloading multiple files at once - you can now download directories as zip files in version [`0.8.21`](https://github.com/jpillora/cloud-torrent#install)

**Comment by bloggervista** [comment]
any updates?
